packages/async-flow/test/types.test-d.ts

```diff
@@ -20,3 +24,48 @@ expectType<(p1: number, p2: number) => Vow<void>>(
   // @ts-expect-error incompatible return type
   null as unknown as HostOf<typeof sumPromise>,
 );
+
+// Test HostInterface and GuestInterface with an exoClass object
+type ExoAPIBase = {
+  getValue: () => number;
+  setValue: (value: number) => void;
+  getCopyData: () => Record<string, number>[];
+};
+type ExoGuestAPI = ExoAPIBase & {
+  getValueAsync: () => Promise<number>;
+};
+
+type ExoHostAPI = ExoAPIBase & {
+  getValueAsync: () => Vow<number>;
+};
+
+expectType<
+  ExoAPIBase & {
+    getValueAsync: () => Vow<number>;
+  }
+>(null as unknown as HostInterface<ExoGuestAPI>);
+expectType<
+  ExoAPIBase & {
+    getValueAsync: () => Promise<number>;
+  }
+>(null as unknown as GuestInterface<ExoHostAPI>);
```

Comment:
I see `null as unknown as Type` was here before but it's getting a little hard to parse. consider,
```ts
const castable: unknown = null;
```

Reply:
Cool will do. Any thoughts on how to mock a remotable? https://github.com/Agoric/agoric-sdk/pull/10045#discussion_r1751074827